src/VariableQueryEditor.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { useState } from 'react';
+import { CassandraVariableQuery } from './models';
+import { TextArea } from '@grafana/ui';
+
+interface VariableQueryProps {
+  query: CassandraVariableQuery;
+  onChange: (query: CassandraVariableQuery, definition: string) => void;
+}
+
+export const VariableQueryEditor = ({ onChange, query }: VariableQueryProps) => {
+  const [state, setState] = useState(query);
+
+  const saveQuery = () => {
+    onChange(state, `${state.rawQuery}`);
+  };
+
+  const handleChange = (event: React.FormEvent<HTMLTextAreaElement>) =>
+    setState({
+      ...state,
+      [event.currentTarget.name]: event.currentTarget.value,
+    });
+
+  return (
+    <>
+        <label className="small">Specify a query that returns variable values. If the query returns one column, it will be interpreted as values, if the query returns two columns, the first one is interpreted as a label (human-readable) and the second one as a value. Due to grafana limitations, both the labels and values must be strings, use `CAST(column as text)` if needed.</label>
```

Comment:
I believe, that moving label on second position would make UX a bit more clear and natural. A tip text could be rewritten like this:

```
Specify a query that returns variable values and, optionally, their labels. Only strings are allowed here, so use CAST(column as text) if needed. First returned column interpreted as a value and second as a label. Labels should be used in cases when there is an intention to hide exact variable values behind human-readable names in UI.
```

Reply:
I like the new text and idea to move values <-> labels!